src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglxnidqinterface.py

```diff
@@ -0,0 +1,137 @@
+        verbose: bool
+            Whether to output verbose text. Default: True.
+        """
+        self.stream_id = "nidq"
+
+        folder_path = Path(file_path).parent
+        super().__init__(folder_path=folder_path, stream_id=self.stream_id, verbose=verbose)
+        self.source_data.update(file_path=str(file_path))
+
+        self.meta = self.recording_extractor.neo_reader.signals_info_dict[(0, self.stream_id)]["meta"]
+
+    def get_metadata_schema(self):
+        metadata_schema = super().get_metadata_schema()
+        metadata_schema["properties"]["Ecephys"]["properties"].update(
+            ElectricalSeriesRaw=get_schema_from_hdmf_class(ElectricalSeries)
+        )
+        return metadata_schema
+
+    def get_metadata(self):
+        metadata = super().get_metadata()
+        session_start_time = get_session_start_time(self.meta)
+        if session_start_time:
+            metadata = dict_deep_update(metadata, dict(NWBFile=dict(session_start_time=session_start_time)))
+
+        # Device metadata
+        metadata = dict(Ecephys=dict())
+        device_key = "niDev1ProductName"  # At least, this is true for the GIN test data
+        if device_key in self.meta:
+            header_metadata = dict(self.meta)
+            for exclude_key in ["fileCreateTime", device_key]:
+                header_metadata.pop(exclude_key)
+            device = dict(
+                name=self.meta[device_key],
+                description=json.dumps(header_metadata),
+                manufacturer="National Instruments",
+            )
+            metadata["Ecephys"].get("Device", list()).append(Device=device)
+
+        # Add groups metadata
+        group_names = self.recording_extractor.get_property("group_name")
+        if self.recording_extractor.get_property("group_name") is None:
+            electrode_groups = [
+                dict(
+                    name="NIDQChannelGroup",
+                    description="A group representing the NIDQ channels.",
+                    location="unknown",
+                    device=device["name"],
+                )
+            ]
+        else:
+            electrode_groups = [
+                dict(
+                    name=group_name,
+                    description="A group representing the NIDQ channels.",
+                    location="unknown",
+                    device=device["name"],
+                )
+                for group_name in set(group_names)
+            ]
+        metadata["Ecephys"].get("ElectrodeGroup", list()).extend(electrode_groups)
+
+        # Electrodes columns descriptions
+        metadata["Ecephys"]["Electrodes"] = [
+            dict(name="group_name", description="Name of the ElectrodeGroup this electrode is a part of."),
+        ]
+
+        metadata["Ecephys"]["ElectricalSeriesSync"] = dict(
+            name="ElectricalSeriesSync", description="Raw acquisition traces for the synchronization (nidq) data."
+        )
+        return metadata
```

Comment:
https://nwbinspector.readthedocs.io/en/dev/best_practices/time_series.html#subtypes

Reply:
That's something we missed in the NeuroNexus review, then, too.

Reply:
This is quite unfortunate as well; there's a TON of metadata information about the configuration of the NIDQ device and since `TimeSeries` can't accept a `Device` link, I guess I'd be forced to dump all that info into the `TimeSeries.description` instead?

I guess I could also make a separate custom `DynamicTable` to serve as the metadata representation for this stuff - downside is it can't be linked to the series. Other suggestions?

For a more general approach though, it's also extremely intuitive for users (and even devs) to want to use something called `ElectricalSeries` to represent their data that is in units `Volts`; if the issue is less about the `ElectricalSeries` neurodata type, and more about how the `electrodes` table (not yet a neurodata type) is only supposed to specify electrode metadata for neurophysiological-related, then the best solution may simply be to make a proper `ElectrodeTable` type and follow it up with a separate child class that can apply to describing environmental electrodes and all their metadata.